caproto/_log.py

```diff
@@ -143,9 +151,111 @@ def color_logs(color):
 
 
 logger = logging.getLogger('caproto')
+ch_logger = logging.getLogger('caproto.ch')
+search_logger = logging.getLogger('caproto.bcast.search')
 current_handler = None  # overwritten below
 
 
+class PVFilter(logging.Filter):
+    '''
+    Pass through only messages relevant to one or more PV names and
+    pv-unrelated message.
+
+    Parameters
+    ----------
+    names : string or list of string
+        PV name or PV name list which will be filtted in.
+
+    Return
+    ------
+    Bool: True or False
+        True if message is not PV related which doesn't has 'pv' as key in extra
+        True if 'pv' as key exist and pv name exist in Filter list.
+        False if message is PV related but pv name isn't in Filter list.
+    '''
+    def __init__(self, names):
+        self.names = names
+    def filter(self, record):
+        if hasattr(record, 'pv'):
+            return record.pv in self.names
+        else:
+            return True
+
+
+class PVOnlyFilter(logging.Filter):
+    '''
+    99% as same as PVFilter class but only default return for PV unrelated message
```

Comment:
The only difference is the put pv unrelated message in or out which is last return True/False. I think `PVOnlyFilter` might be abandoned since I put a extra key word argument in `PVFilter` which could realize same goal.

Reply:
I see. Thanks for explaining.